core/src/types/builder.rs

```diff
@@ -40,9 +41,18 @@ pub trait Builder: Default {
     const SCHEME: Scheme;
     /// The accessor that built by this builder.
     type Accessor: Access;
+    /// The config for this builder.
+    type Config: DeserializeOwned;
+
+    /// Construct a builder from given config.
+    fn from_config(config: Self::Config) -> Self;
```

Comment:
I want to return `Result<Self>` in `from_config` and `from_map` to avoid panic during the build process. Do you want to implement in this PR or start a new PR for this?

Reply:
Let's make further non-blocker change in a follow-up PR. I can do it, but not in this one unless it's necessary.